src/Validation/Validator.php

```diff
@@ -160,14 +160,26 @@ public function __construct()
     }
 
     /**
-     * Returns an array of fields that have failed validation. On the current model. This method will
-     * actually run validation rules over data, not just return the messages.
+     * Validates and returns an array of failed fields and their error messages.
      *
      * @param array $data The data to be checked for errors
      * @param bool $newRecord whether the data to be validated is new or to be updated.
-     * @return array Array of invalid fields
+     * @return array Array of failed fields
+     * @deprecated 3.9.0 Renamed to validate()
      */
     public function errors(array $data, $newRecord = true)
+    {
+        return $this->validate($data, $newRecord);
```

Comment:
This probably should trigger deprecation notes.
If your code extends this method in plugins or project, it will otherwise silently not run anymore right?

Reply:
deprecation warning can be add to 4.x 
and then remove the old methods in cake 5

Reply:
Do we also backport to 3.9? Or keep as is?

Reply:
we should not add more deprecation warning to cake 3.x ( when no needed ) 
because we are in RC(cake 4) level and we will see final version soon 
so people should not force to upgrade their apps to 3.9 IMO ( that there is not yet) 
I think 3.8 is enough 
but 3.9 can be receive backport features from 4.x ( until 3.9 get out )

having this method ( `errors` ) does not heart code and can be kept as an alias to new method

Reply:
if we merge this now , this method must throw deprecation warning in cake 4